libs/xml/tinyxml.cpp

```diff
@@ -1141,7 +1152,13 @@ void TiXmlAttribute::SetIntValue( int _value )
 void TiXmlAttribute::SetDoubleValue( double _value )
 {
 	char buf [64];
-	sprintf (buf, "%lf", _value);
+	std::stringstream sst;
+	sst.imbue(std::locale::classic());
+	sst << std::fixed;
+	sst << std::showpoint;
+	sst << std::setprecision(6);
```

Comment:
Why 6? Doubles have up to 13 points of precision. Perhaps removing the fixed, showpoint, and precision is best to match the original %lf behavior; but if you want to keep them at least setting the precision higher than 6 seems appropriate for a double.

Reply:
6 is default value of sprintf, and presets this far have used that.

Reply:
Without fixed and showpoint instead of value="1.000000" we will get value="1"
Sure functionally that is same.

Reply:
And for higher precision, I would expect that more than 6 decimals would cause any audible difference in any parameter :-)

Reply:
OK that’s a pretty good argument about keeping same behavior. Sure great.

If the automated builds work, I’ll merge this.

If you’d like, please add yourself as you would like to be attributed in the AUTHORS file and send in a pull request. We are keeping it alphabetical but happy to have you added!

Thanks again